buku

```diff
@@ -521,8 +538,7 @@ class BukuDb:
             A list of tuples representing bookmark records.
         """
 
-        self.cur.execute('SELECT * FROM bookmarks')
-        return self.cur.fetchall()
+        return self._fetch('SELECT * FROM bookmarks')
```

Comment:
How is this indirection helping? What was the problem with the earlier method?

Reply:
In an interpreted language, isn't this slower?

Reply:
> How is this indirection helping? What was the problem with the earlier method?

Repeating multiple times nearly identical sequences of statements are boilerplate code – it's not doing anyone any favours, bloating the code, and overall introducing potential issues whenever the sequence is repeated imperfectly (the likelihood for which grows whenever a new instance of the boilerplate is added); not to mention when this behaviour needs to be updated in every instance (not exactly a rare occurrence either). Avoiding them is basis for healthy codebase (even moreso when the library alone is nearly 200Kb in a single file).

Rather, it's odd that nobody seems to have thought to combine these two actions in the database API (or at least in Python version) – it seems like an incredibly common usage scenario to me.

> In an interpreted language, isn't this slower?

The speed difference is completely negligible – it's not an action you're going to evoke hundreds of thousands of times per second with real-time constraints (…and even if it was, processing of the database query itself would take up much more computation than a measly extra function call, scripting language or not).

Reply:
We are talking of a very simple standard sequence to fetch DB results here on a stable codebase. However, I am fine with this since it saves lines of code.